lib/http/timeout/per_operation.rb

```diff
@@ -59,14 +59,17 @@ def write(data)
         # Read data from the socket
         def readpartial(size)
           loop do
-            result = socket.read_nonblock(size, :exception => false)
+            result = rescue_readable do
```

Comment:
If you recall from before, JRuby silently discards `:exception => false` on SSL sockets and still throws exceptions. I can add a comment to that effect.